lib/Transforms/MapArithToComb.cpp

```diff
@@ -0,0 +1,135 @@
+//===- MapArithToComb.cpp - Arith-to-comb mapping pass ----------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Contains the definitions of the MapArithToComb pass.
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetail.h"
+#include "circt/Dialect/Comb/CombOps.h"
+#include "circt/Dialect/HW/HWOpInterfaces.h"
+#include "circt/Dialect/HW/HWOps.h"
+#include "circt/Transforms/Passes.h"
+#include "mlir/Conversion/LLVMCommon/ConversionTarget.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+using namespace mlir;
+using namespace circt;
+
+template <typename TFrom, typename TTo>
+class OneToOnePattern : public OpConversionPattern<TFrom> {
+public:
+  OneToOnePattern(MLIRContext *context) : OpConversionPattern<TFrom>(context) {}
+  using OpAdaptor = typename TFrom::Adaptor;
+
+  LogicalResult
+  matchAndRewrite(TFrom op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    rewriter.replaceOpWithNewOp<TTo>(op, adaptor.getOperands(), op->getAttrs());
+    return success();
+  }
+};
+
+class ExtSConvertionPattern : public OpConversionPattern<arith::ExtSIOp> {
+public:
+  ExtSConvertionPattern(MLIRContext *context)
+      : OpConversionPattern<arith::ExtSIOp>(context) {}
+  using OpAdaptor = typename arith::ExtSIOp::Adaptor;
+
+  LogicalResult
+  matchAndRewrite(arith::ExtSIOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    size_t outWidth = op.getType().getIntOrFloatBitWidth();
```

Comment:
`arith` also supports vectors and tensors in which case this would just crash. Maybe add a `TypeConverter` which only marks `IntegerType` legal or check it otherwise?

Reply:
Good catch - we definitely want to make it clear what things can and can't be trivially converted with this pass.